consensus/core/src/dag_state.rs

```diff
@@ -79,30 +79,72 @@ impl DagState {
     /// Accepts a block into DagState and keeps it in memory.
     pub(crate) fn accept_block(&mut self, block: VerifiedBlock) {
         let block_ref = block.reference();
+        // Ensure we don't write multiple blocks per slot for our own index
+        if block_ref.author == self.context.own_index {
+            let existing_blocks = self.get_blocks_at_slot(Slot::from(block_ref));
+            if !existing_blocks.is_empty() {
+                // TODO: should we panic?
```

Comment:
It would be fine if the existing block has the same ref, but still let's panic here. Otherwise the check would not be useful, e.g. the block can proceed to get broadcasted.